test/e2e/gc_e2e_test.go

```diff
@@ -33,159 +32,281 @@ var _ = Describe("Garbage collector", func() {
-		}
-
-		// add owners
-		ownerutil.AddOwner(dependent, fetchedA, true, false)
-		ownerutil.AddOwner(dependent, fetchedB, true, false)
-
-		// create dependent
-		_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Create(dependent)
-		require.NoError(GinkgoT(), err, "dependent could not be created")
-
-		// delete ownerA in the foreground (to ensure any "blocking" dependents are deleted before ownerA)
-		propagation := metav1.DeletionPropagation("Foreground")
-		options := metav1.DeleteOptions{PropagationPolicy: &propagation}
-		err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(fetchedA.GetName(), &options)
-		require.NoError(GinkgoT(), err)
-
-		// wait for deletion of ownerA
-		waitForDelete(func() error {
-			_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(ownerA.GetName(), metav1.GetOptions{})
-			return err
 		})
+	})
+
+	// TestOwnerReferenceGCBehavior runs a simple check on OwnerReference behavior to ensure
+	// a resource with multiple OwnerReferences will not be garbage collected when one of its
+	// owners has been deleted.
+	// Test Case:
+	//				CSV-A     CSV-B                        CSV-B
+	//				   \      /      --Delete CSV-A-->       |
+	//				   ConfigMap						 ConfigMap
+	Describe("Multiple OwnerReferences will not be garbage collected", func() {
+
+		Context("Creating multiple owners", func() {
+			var (
+				ownerA   v1alpha1.ClusterServiceVersion
+				ownerB   v1alpha1.ClusterServiceVersion
+				fetchedA *v1alpha1.ClusterServiceVersion
+				fetchedB *v1alpha1.ClusterServiceVersion
+				err      error
+			)
+			BeforeEach(func() {
+				ownerA = newCSV("ownera", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("dep-", nil, nil))
+				ownerB = newCSV("ownerb", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("dep-", nil, nil))
+
+				// create all owners
+				fetchedA, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(&ownerA)
+				Expect(err).NotTo(HaveOccurred())
+				fetchedB, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(&ownerB)
+				Expect(err).NotTo(HaveOccurred())
+			})
+			Context("and associating them to a dependent", func() {
+
+				var (
+					dependent *corev1.ConfigMap
+				)
+
+				BeforeEach(func() {
+					dependent = &corev1.ConfigMap{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "dependent",
+						},
+						Data: map[string]string{},
+					}
+
+					// add owners
+					ownerutil.AddOwner(dependent, fetchedA, true, false)
+					ownerutil.AddOwner(dependent, fetchedB, true, false)
+
+					// create ConfigMap dependent
+					_, err = kubeClient.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Create(dependent)
+					Expect(err).NotTo(HaveOccurred(), "dependent could not be created")
+				})
+
+				Context("owner reference behavior can be verified", func() {
```

Comment:
shouldn't this describe the context more? something about the foreground propogation policy?

Reply:
I have added it :)